src/types/AST.js

```diff
@@ -3,7 +3,7 @@
 import type { i32 } from "../interpreter/runtime/values/i32";
 
 type U32Literal = NumberLiteral & {
-  value: NumberInterface<i32>
+  value: NumericOperations<i32>
```

Comment:
This doesn't look right to me, this interface defines operations that can be performed on a numeric type. I don't think that is appropriate for the AST, which I see as just a data representation.

Reply:
This is not clear enough, could we discuss about that on Monday next week?

Reply:
yeah, no worries